app/src/main/java/de/s3xy/architecturesample/github/GithubInteractor.java

```diff
@@ -0,0 +1,81 @@
+package de.s3xy.architecturesample.github;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import de.s3xy.architecturesample.github.model.RepositoriesSearchResult;
+import de.s3xy.architecturesample.network.NetworkConnectivityManager;
+import de.s3xy.architecturesample.network.NetworkModule;
+import de.s3xy.architecturesample.network.NoNetworkException;
+import rx.Observable;
+
+/**
+ * Created by Vlad Fedorenko <vfedo92@gmail.com>
+ * 08.26.2016
+ */
+public class GithubInteractor {
+    private GithubApi mGithubAuthApi;
+    private GithubApi mGithubApi;
+    private GithubAuthManager mGithubAuthManager;
+    private NetworkConnectivityManager mNetworkConnectivityManager;
+
+    @Inject
+    GithubInteractor(@Named(NetworkModule.AUTH_FLOW) GithubApi githubAuthApi,
+                     GithubApi api,
+                     GithubAuthManager authManager,
+                     NetworkConnectivityManager networkConnectivityManager) {
+        mGithubAuthApi = githubAuthApi;
+        mGithubApi = api;
+        mGithubAuthManager = authManager;
+        mNetworkConnectivityManager = networkConnectivityManager;
+    }
+
+    public boolean shouldAuthorize() {
+        return !mGithubAuthManager.shouldBeUnauthorized() && mGithubAuthManager.getAccessToken() == null;
+    }
+
+    public Observable<Boolean> getAccessToken(String code) {
+        if (!mNetworkConnectivityManager.isOnline()) {
```

Comment:
You a right, I'll fix it. But I believe that flatMap is more suitable here, because we don't need inline code that compose provides, we should react to an item in the chain. Am I right?